plugins/aggregators/starlark/starlark.go

```diff
@@ -0,0 +1,143 @@
+package starlark //nolint - Needed to avoid getting import-shadowing: The name 'starlark' shadows an import name (revive)
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+	common "github.com/influxdata/telegraf/plugins/common/starlark"
+	"go.starlark.net/starlark"
+)
+
+const (
+	description  = "Aggregate metrics using a Starlark script"
+	sampleConfig = `
+  ## The Starlark source can be set as a string in this configuration file, or
+  ## by referencing a file containing the script.  Only one source or script
+  ## should be set at once.
+  ##
+  ## Source of the Starlark script.
+  source = '''
+state = {}
+
+def add(metric):
+  state["last"] = metric
+
+def push():
+  return state.get("last")
+
+def reset():
+  state.clear()
+'''
+
+  ## File containing a Starlark script.
+  # script = "/usr/local/bin/myscript.star"
+
+  ## The constants of the Starlark script.
+  # [aggregators.starlark.constants]
+  #   max_size = 10
+  #   threshold = 0.75
+  #   default_name = "Julia"
+  #   debug_mode = true
+`
+)
+
+type Starlark struct {
+	common.StarlarkCommon
+}
+
+func (s *Starlark) Init() error {
+	// Execute source
+	err := s.StarlarkCommon.Init()
+	if err != nil {
+		return err
+	}
+
+	// The source should define an add function.
+	err = s.AddFunction("add", &common.Metric{})
+	if err != nil {
+		return err
+	}
+
+	// The source should define a push function.
+	err = s.AddFunction("push")
+	if err != nil {
+		return err
+	}
+
+	// The source should define a reset function.
+	err = s.AddFunction("reset")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Starlark) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Starlark) Description() string {
+	return description
+}
+
+func (s *Starlark) Add(metric telegraf.Metric) {
+	parameters, found := s.GetParameters("add")
+	if !found {
+		s.Log.Errorf("The parameters of the add function could not be found")
+		return
+	}
+	parameters[0].(*common.Metric).Wrap(metric)
+
+	_, err := s.Call("add")
+	if err != nil {
+		s.LogError(err)
+	}
+}
+
+func (s *Starlark) Push(acc telegraf.Accumulator) {
+	rv, err := s.Call("push")
+	if err != nil {
+		s.LogError(err)
```

Comment:
Maybe we should use `acc.AddError` here to allow for statistics? Same for the instances below.

Reply:
The accumulator is not available in case of the other instances